viewing_party/party.py

```diff
@@ -1,23 +1,171 @@
 # ------------- WAVE 1 --------------------
+"""Create and modify movie data"""
+
 
 def create_movie(title, genre, rating):
-    pass
+    """create a movie dictionary"""
+    if title is None or genre is None or rating is None:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
+
+
+def add_to_watched(user_data, movie):
+    """append movie dictionary to user_data "watched" list"""
+    user_data["watched"].append(
+        create_movie(movie["title"], movie["genre"], movie["rating"])
+    )
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    """append movie dictionairy to user_data "watchlist" list"""
+    user_data["watchlist"].append(
+        create_movie(movie["title"], movie["genre"], movie["rating"])
+    )
+    return user_data
+
+
+def watch_movie(user_data, title):
+    """Move a movie from user_data "watchlist" list to "watched" list"""
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    """Return the user's average movie rating"""
+    rating = 0
+    try:
+        for movie in user_data["watched"]:
+            rating += movie["rating"]
+        return rating / len(user_data["watched"])
+    except ZeroDivisionError:
+        return 0.0
+
+
+def get_most_watched_genre(user_data):
+    """Return the user's most frequently watched genre"""
+    genre_list = []
+    try:
+        for movie in user_data["watched"]:
+            genre_list.append(movie["genre"])
+
+        sorted_genre_list = sorted(genre_list, key=genre_list.count, reverse=True)
+        return sorted_genre_list[0]

```

Comment:
[perf] What's the big-O time complexity of this? If you didn't have built-in functions, how might you find the most-frequent genre in the data?